src/main/kotlin/no/nav/klage/client/StsClient.kt

```diff
@@ -0,0 +1,58 @@
+package no.nav.klage.client
+
+import brave.Tracer
+import com.fasterxml.jackson.annotation.JsonProperty
+import no.nav.klage.util.getLogger
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.bodyToMono
+import java.time.LocalDateTime
+
+@Component
+class StsClient(private val stsWebClient: WebClient, private val tracer: Tracer) {
+
+    private var cachedOidcToken: Token? = null
+
+    companion object {
+        @Suppress("JAVA_CLASS_ON_COMPANION")
+        private val logger = getLogger(javaClass.enclosingClass)
+    }
+
+    @Value("\${navCallId}")
+    lateinit var navCallId: String
+
+    fun oidcToken(): String {
+        if (cachedOidcToken.shouldBeRenewed()) {
+            logger.debug("Getting token from STS")
+            cachedOidcToken = stsWebClient.get()
+                .uri { uriBuilder ->
+                    uriBuilder
+                        .queryParam("grant_type", "client_credentials")
+                        .queryParam("scope", "openid")
+                        .build()
+                }
+                .header(navCallId, tracer.currentSpan().context().traceIdString())
+                .retrieve()
+                .bodyToMono<Token>()
+                .block()
+        }
+
+        return cachedOidcToken!!.token
+    }
+
+    private fun Token?.shouldBeRenewed(): Boolean = this?.hasExpired() ?: true
+
+    data class Token(
```

Comment:
Jeg har lagt inn en klasse OidcToken i domain som kan brukes i stedet for denne.

Reply:
Fint. Bruker den i stedet nå.